nomad/server_test.go

```diff
@@ -319,7 +319,61 @@ func TestServer_Reload_TLSConnections_TLSToPlaintext_RPC(t *testing.T) {
 
 	err := s1.reloadTLSConnections(newTLSConfig)
 	assert.Nil(err)
-	assert.True(s1.config.TLSConfig.Equals(newTLSConfig))
+	assert.True(s1.config.TLSConfig.CertificateInfoIsEqual(newTLSConfig))
+
+	codec := rpcClient(t, s1)
```

Comment:
This won't actually use TLS: https://github.com/hashicorp/nomad/blob/40db0af310d13545429357bed7dcc28f35d8f6ed/nomad/rpc_test.go#L27-L36

Reply:
You may want to add a new helper that takes a tls config and then returns a client codec. You would have to write the TLS mode first:

```
conn.Write([]byte{byte(pool.RpcTLS)}) 
conn.Write([]byte{byte(pool.RpcNomad)}) 
```

You can remove the new code and put a TODO for yourself, and create a follow up PR